### ISSUE TITLE
neovim: `latexFormatter = "latexindent"` add newline on buffer save

### ISSUE THREAD

**Issue by mangkoran** [latex-lsp/texlab#895]
First of all, thank you for making such great LSP for latex!

I encountered an issue on my texlab setup. I am using `neovim` with `nvim-lspconfig` and `vimtex`. If I enable `texlab` LSP in `nvim-lspconfig`, there will be newline added on every time I save a buffer. It adds the newline even though I didn't make any changes to the buffer. If I disable `texlab` in `nvim-lspconfig`, the issue will be gone.

I have little to none knowledge whether this is `texlab` issue or `nvim-lspconfig` issue. Please CMIIW!

**Comment by clason** [comment]
I cannot reproduce that. Please provide a minimal reproducing config.

**Comment by mangkoran** [comment]
Apologize, but I am encountering difficulties while trying to provide minimal reproducible example. I tried to create MRE by using Arch container image and did the following step:
1. Install `git, neovim`
2. Install LazyVim
```
git clone https://github.com/LazyVim/starter ~/.config/nvim
```
3. Run `nvim` once for LazyVim to update
4. Setup `nvim-lspconfig` with `texlab` LSP
```lua
-- in ~/.config/nvim/lua/plugins/lsp.lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        texlab = {},
      },
    },
  },
}
```
6. Quit and run `nvim foo.tex`. `mason` will try to install `texlab` then try `:LspStart` and save the buffer. 

However, I did not encounter the newline issue within the Arch container setup above.

Then, I tried to nuke my main `neovim` setup, tried to configure LazyVim following the steps above. Interestingly, the issue _still_ persist although I have nuke my entire config.

**Comment by Veirt** [comment]
Hello, I can reproduce it by setting the `latexFormatter` to `"latexindent"`.

https://github.com/latex-lsp/texlab/assets/55097092/0114d7e7-071a-4205-966e-e233daebd32f

latexindent version: `3.20.1, 2023-01-07`
My old config:
```lua
require("lspconfig").texlab.setup({
    settings = {
        texlab = {
            latexFormatter = "latexindent",
        },
    },
})
```

Changing the `latexFormatter` to `"texlab"` (that is not yet implemented, as I've read [here](https://github.com/latex-lsp/texlab/wiki/Configuration#texlablatexindentlocal) solved my issue from adding new line.

**Comment by mangkoran** [comment]
@Veirt Thank you very much. I can also confirm this behavior.
```
latexindent: 3.20.3, 2023-02-19
```

**Comment by Veirt** [comment]
You're welcome @mangkoran :)

**Comment by pfoerster** [comment]
This should be fixed now with 365c04050c7dd871f815941ca616e9373d5f475a (the range of the `TextEdit` was too small and did not include the final newline in the source file).

**Comment by mangkoran** [comment]
I can confirm the fix above. First half is `texlab 5.6.0`. Second half is `texlab-git r2153.365c0405-1` compiled from AUR.
https://github.com/latex-lsp/texlab/assets/57625126/4920a1b9-2712-4232-b7cd-cdc374469f01
